platform/darwin/src/MGLCircleStyleLayer.mm

```diff
@@ -47,6 +47,14 @@ - (instancetype)initWithIdentifier:(NSString *)identifier source:(MGLSource *)so
     return self;
 }
 
+- (instancetype)initWithRawLayer:(mbgl::style::CircleLayer *)rawLayer source:(MGLSource *)source
+{
+    if (self = [super initWithIdentifier:@(rawLayer->getID().c_str()) source:source]) {
```

Comment:
Ah, you’re right – I mistakenly thought the `mbgl::style::Layer` has a method to get the `mbgl::style::Source`, but of course it all comes from the `mbgl::Map` by ID.